File: Basics/OOPS/CPP/DebitCard.cpp
#include "Card.cpp"
using namespace std;

class DebitCard : public Card{
    public: 
    DebitCard(const int &number, const string &name) : Card(number, name){}
    void processPayment(double amount) override{
        cout << "Processing debit card payment of $" << amount << " for card holder: " << getCardHolderName() << endl;
    }
};

File: Basics/OOPS/JAVA/DebitCard.java
public class DebitCard extends Card {
    public DebitCard(String cardNumber, String cardHolderName) {
        super(cardNumber, cardHolderName);
    }

    @Override
    public void pay(double amount) {
        System.out.println("Paid " + amount + " using Debit Card: " + getCardHolderName() + " with card number " + getCardNumber());
    }
}